superset-frontend/packages/superset-ui-core/src/query/normalizeTimeColumn.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import omit from 'lodash/omit';
+
+import {
+  AdhocColumn,
+  isAdhocColumn,
+  isPhysicalColumn,
+  QueryFormColumn,
+  QueryFormData,
+  QueryObject,
+} from './types';
+import { FeatureFlag, isFeatureEnabled } from '../utils';
+
+export function normalizeTimeColumn(
+  formData: QueryFormData,
+  queryObject: QueryObject,
+): QueryObject {
+  if (!isFeatureEnabled(FeatureFlag.GENERIC_CHART_AXES)) {
+    return queryObject;
+  }
+  if (isFeatureEnabled(FeatureFlag.GENERIC_CHART_AXES) && !formData.x_axis) {
+    return queryObject;
+  }
+
+  const mutatedColumns: QueryFormColumn[] = [...(queryObject.columns || [])];
+  const axisIdx = queryObject.columns?.findIndex(
```

Comment:
IMO, avoiding redundant local variables is for readability, in this case, readability refers to the intermediate variables that do not bring benefits but add cognitive load -- I have to remember that `columns` and `extra` is from `queryObject` other than otherwhere.

Reply:
This can be very subjective and different for different people, but in general I find narrower and longer code snippets more readable than wider and shorter. In this case I find @michael-s-molina 's destructuring approach more readable. Having said that I'm ok with both approaches, as long as the function/file follows a similar pattern throughout.

Reply:
Thanks both.  I have changed to `const { columns: _columns, extras: _extras } = queryObject;`, It's hard to say which approach is better, but the ease of tracking is always my first concern. (especially, in the conversion function)